radix-engine/src/transaction/transaction_receipt.rs

```diff
@@ -84,10 +75,17 @@ impl TransactionReceipt {
         }
     }
 
-    pub fn expect_commit_success(&self) -> &Vec<Vec<u8>> {
+    pub fn expect_rejection(&self) -> &RejectionError {
+        match &self.result {
+            TransactionResult::Commit(..) => panic!("Expected rejection but was commit"),
+            TransactionResult::Reject(ref r) => &r.error,
+        }
+    }
+
+    pub fn expect_commit_success(&self) -> &[u8] {
         match &self.result {
             TransactionResult::Commit(c) => match &c.outcome {
-                TransactionOutcome::Success(x) => x,
+                TransactionOutcome::Success(x) => &x[1][..],
```

Comment:
I don't think it's proper to assume the output of the second instruction is the expected output. With Radix transaction model, a transaction may include "any" number of instructions and developers need to be able to inspect output of any instruction.

Suggest keeping the original `expect_commit_success` and change `output()` to `output(nth: usize)`.